File: DSA/Ayush_Mishra_590014844/Day_41_Question_1.c
#include <stdio.h>
#include <string.h>

void removeAdjacentDuplicates(char* s) {
    char stack[1000];
    int top = -1;
    for (int i = 0; s[i]; i++) {
        if (top >= 0 && stack[top] == s[i]) top--;
        else stack[++top] = s[i];
    }
    stack[++top] = '\0';
    printf("%s\n", stack);
}

int main() {
    char str[1000];
    printf("Enter the string: ");
    scanf("%s", str);
    removeAdjacentDuplicates(str);
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_41_Question_1.py
def remove_adjacent_duplicates(s):
    stack = []
    for ch in s:
        if stack and stack[-1] == ch:
            stack.pop()
        else:
            stack.append(ch)
    return ''.join(stack)

s = input("Enter the string: ")
print(remove_adjacent_duplicates(s))
